src/com/cft/contactmerge/FirstNameMatcher.java

```diff
@@ -0,0 +1,42 @@
+/* Created by James Page on May 8th, 2018
+ * Contains the order in which the first names will be considered a match or not...
+ * Exact Match (and matched with punctuations), matched with hyphenated parts, matched with apostrophe, and matched with initial.
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+
+public abstract class FirstNameMatcher extends NameMatchLogic {
```

Comment:
Since everything is static, we don't need "extends NameMatchLogic", right? Nothing is shared or overridden in these class.

Reply:
So would you rather import it as a package? I just want to avoid instantiating objects for every match checked.

Reply:
For the time being I'd keep them static but remove the "extends" and make the methods public so that you can test them.

Reply:
Just making them public isn't enough. I can do one of two things...
1: add the line of code "import static com.cft.contactmerge.NameMatchLogic.*;" at the top of every class.
2. since the methods are static, I can call the class and then the method, for example... "NameMatchLogic.isNamesExactMatch(contactOneLastName, contactTwoLastName)"

Which way do you think is better?

Reply:
I don't have a preference. I'm use to providing the class name when I call static methods but I get the sense that "import static" is a technique that Java encourages people to use.